apps/api-extractor/src/generators/DeclarationReferenceGenerator.ts

```diff
@@ -279,10 +269,33 @@ export class DeclarationReferenceGenerator {
   }
 
   private _getParentReference(symbol: ts.Symbol): DeclarationReference | undefined {
-    // First case: the symbol is exported via a namespace
-    const exportingNamespace: ts.Symbol | undefined = this._collector.getExportingNamespace(symbol);
-    if (exportingNamespace) {
-      return this._symbolToDeclarationReference(exportingNamespace, exportingNamespace.flags, true);
+    const declaration: ts.Node | undefined = TypeScriptHelpers.tryGetADeclaration(symbol);
+    const sourceFile: ts.SourceFile | undefined = declaration?.getSourceFile();
+
+    const entity: CollectorEntity | undefined = this._collector.tryGetEntityForSymbol(symbol);
+    if (entity) {
+      if (entity.exportedFromEntryPoint) {
+        return new DeclarationReference(this._sourceFileToModuleSource(sourceFile));
+      }
+
+      const firstExportingConsumableParent: CollectorEntity | undefined =
+        entity.firstExportingConsumableParent;
+      if (
+        firstExportingConsumableParent &&
+        firstExportingConsumableParent.astEntity instanceof AstNamespaceImport
+      ) {
+        const parentSymbol: ts.Symbol | undefined = TypeScriptInternals.tryGetSymbolForDeclaration(
+          firstExportingConsumableParent.astEntity.declaration,
+          this._collector.typeChecker
+        );
+        if (parentSymbol) {
+          return this._symbolToDeclarationReference(
+            parentSymbol,
+            parentSymbol.flags,
+            /*includeModuleSymbols*/ true
+          );
+        }
+      }
     }
```

Comment:
This is basically what's being discussed in https://github.com/microsoft/rushstack/pull/3602#discussion_r965327458, but I think the ambiguity is a bit more explicit now, which is a good thing.